### ISSUE TITLE
`--kubeconfig` doesn't work

### ISSUE THREAD

**Issue by amitsaha** [aquasecurity/kubectl-who-can#41]
Hi, passing a custom kubeconfig file path via `--kubeconfig` doesn't work. Setting `KUBECONFIG` env var works. For example:

Doesn't work:

```
./kubectl-who-can --kubeconfig=/home/amit/.kube/k8s-cluster-non-production get secret mytoken -n myns
Error: resolving resource: the server doesn't have a resource type "secret"
```

Works: 
```
$ KUBECONFIG=/home/amit/.kube/k8s-cluster-non-production ./kubectl-who-can get secret mytoken -n myns

ROLEBINDING           NAMESPACE   SUBJECT  TYPE            SA-NAMESPACE
....
CLUSTERROLEBINDING                                    SUBJECT                             TYPE            SA-NAMESPACE
..
```

**Comment by danielpacak** [comment]
Hi @amitsaha . Thank you for reporting this. I was trying to reproduce it on my end with no luck, i.e., who-can allows me to  switch configs with the --kubeconfig flag.

What is not handled properly though is the error message when the kubeconfig and the current context points to a server that cannot be reached. who-can says that a give resource cannot be found, whereas it should report a connection error.

That said, could you confirm that the following kubectl commands work for you?

```
kubectl --kubeconfig=/home/amit/.kube/k8s-cluster-non-production get role -A
```

Also note that the syntax for specifying a resource names is as follows:

```
./kubectl-who-can --kubeconfig=/home/amit/.kube/k8s-cluster-non-production get secret/mytoken -n myns
```

**Comment by amitsaha** [comment]
hi @danielpacak thanks for that. My kubectl version is `1.12.7` - i am using a non-latest version due to the server version of my cluster. Hence, `-A` flag doesn't seem to work for me. However, I tried this (assuming that's what the -A flag does):

```
 kubectl --kubeconfig=/home/amit/.kube/k8s-cluster-non-production get role --all-namespaces
NAMESPACE       NAME                                             AGE
ingress-nginx   nginx-ingress-role                               11d
kube-public     system:controller:bootstrap-signer               15d
kube-system     extension-apiserver-authentication-reader        15d
kube-system     system::leader-locking-kube-controller-manager   15d
kube-system     system::leader-locking-kube-scheduler            15d
kube-system     system:controller:bootstrap-signer               15d
kube-system     system:controller:cloud-provider                 15d
kube-system     system:controller:token-cleaner                  15d
...

```

Next, I tried `kubectl-who-can`, but get the same resource not found error:

```
$ ./kubectl-who-can --kubeconfig=/home/amit/.kube/k8s-cluster-non-production get secret/mytoken -n myns
Error: resolving resource: the server doesn't have a resource type "secret"
```

My versions:

```
$ kubectl --kubeconfig=/home/amit/.kube/k8s-cluster-non-production version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.7", GitCommit:"6f482974b76db3f1e0f5d24605a9d1d38fad9a2b", GitTreeState:"clean", BuildDate:"2019-03-25T02:52:13Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.10-eks-2e569f", GitCommit:"2e569fd887357952e506846ed47fc30cc385409a", GitTreeState:"clean", BuildDate:"2019-07-25T23:13:33Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
```

Let me know if I can help you debug this further.

**Comment by danielpacak** [comment]
Thank your for sending more details @amitsaha . I tried reproducing it one more time and I think I found the problem:
- https://github.com/aquasecurity/kubectl-who-can/blob/master/pkg/cmd/list.go#L182 - this line is called too late, i.e. after the client config is already created - which effectively means that the `--kubeconfig` flag is ignored
- If neither `KUBECONFIG` nor `--kubeconfig` are specified the client config defaults to the server at http://localhost:8080
- When a default server is not reachable then obviously kubect-who-can cannot resolve a resource specified as command line args. This message is misleading and we should display connection error instead

I'll fix it and release shortly.

**Comment by amitsaha** [comment]
Ah, glad you found the issue. Thanks, will keep an eye out for the fix.

**Comment by danielpacak** [comment]
> Ah, glad you found the issue. Thanks, will keep an eye out for the fix.

@amitsaha Just to let you know, we've merged the fix to master and it's available in release [v0.1.0-beta.2](https://github.com/aquasecurity/kubectl-who-can/releases/tag/v0.1.0-beta.2).

You can install it with `kubectl krew install who-can`. If you have a previous version installed with krew, just run `kubectl krew upgrade`.

**Comment by amitsaha** [comment]
hi @danielpacak i tried, and looks good. Thank you both!
